src/main/java/com/commercetools/sync/services/impl/TypeServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.commercetools.sync.services.impl;
+
+
+import com.commercetools.sync.services.TypeService;
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.queries.QueryExecutionUtils;
+import io.sphere.sdk.types.queries.TypeQuery;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+/**
+ *TODO: USE graphQL to get only keys
+ *TODO: UNIT TEST
+ *TODO: JAVA DOC
+ */
+public class TypeServiceImpl implements TypeService {
+    private final BlockingSphereClient ctpClient;
+    private final Map<String, String> cache = Collections.synchronizedMap(new HashMap<>());
+
+    public TypeServiceImpl(@Nonnull final BlockingSphereClient ctpClient) {
+        this.ctpClient = ctpClient;
+    }
+
+    @Nullable
+    @Override
+    public String getCachedTypeKeyById(@Nullable final String id) {
```

Comment:
Do you want to achieve thread safety here? If yes, I think this method is not thread safe. It can happen that `fetchAllTypesKeysIntoCache` will be called twice.